R/teal_data-get_code.R

```diff
@@ -39,10 +39,10 @@
 #'     foo()
 #'     y <- x
 #'   })
-#' get_code(data, datanames = "y")
+#' teal.code::get_code(data, names = "y")
```

Comment:
I'm a bit unsure about those changes, to use `teal.code::get_code` directly. But if we don't do that, old `teal.data::get_code` will throw warnings

Reply:
I wonder how `get_code(data, names = "y")` will work for `data` of class `teal_data`.
It will use the `teal.data::get_code` and will treat `names = "y"` as a part of `...`. But `get_code_dependency` will not be used as it's only activated when `datanames` is not `NULL`. So `names` will be ignored.

That's why I put `teal.code::` prefix so it will use the method for `qenv` where `names` is passed to `teal.code::get_code_dependency`.

We either introduce `names` in `teal.data::get_code` or we prefix examples and documentation with `teal.code::`

Reply:
Oh, if we reimplement it this way https://github.com/insightsengineering/teal.data/pull/343#pullrequestreview-2375091411

Then there will be a deprecation warning while using `get_code` on `teal_data` when `teal.data` is attached

Reply:
Maybe deprecation warning is fine, if we remove examples and tests? Then nothing will be displayed in R CMD CHECK as a warning

Reply:
> Oh, if we reimplement it this way [#343 (review)](https://github.com/insightsengineering/teal.data/pull/343#pullrequestreview-2375091411)
> 
> Then there will be a deprecation warning while using `get_code` on `teal_data` when `teal.data` is attached

Good point. Solution is to change a signature and only run this method when `signature = list(object = "teal_data", datanames = "character")`. Only then this method will be run.

Reply:
Makes sense. Then `if` is needed, we don't want to include `datanames` in the generic (not worth it).